### PR TITLE
Sanity checks on offchain tx flow

### PR DESCRIPTION
**Pull request by altafan**
Please @louisinger @Kukks review.

## Summary by CodeRabbit

* **New Features**
  * Added duplicate off-chain transaction detection with explicit error response.

* **Bug Fixes**
  * Enforced stricter state-transition guards to prevent invalid or out-of-order transitions.
  * Persisted events only when actual state changes occur.
  * Finalized transactions consistently mark completion and clear failure reasons.
  * Database retrieval now returns explicit stage flags for failed/ended states.

* **Tests**
  * Added replay and duplicate scenario tests covering edge-case transitions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: internal/core/application/service.go [hunk 1]
```diff
@@ -472,6 +474,19 @@ func (s *service) SubmitOffchainTx(
...
+		log.WithError(err).Errorf("failed to fetch offchain tx %s", txid)
+		return nil, errors.INTERNAL_ERROR.New("something wen wrong").
+			WithMetadata(map[string]any{"txid": txid})
+	}
+
+	if existingOffchainTx != nil {
+		return nil, errors.INVALID_ARK_PSBT.New(
+			"duplicated offchain tx %s", txid,
+		).WithMetadata(errors.PsbtMetadata{Tx: signedArkTx})
+	}
```
[the remark this answers is not in the document]
Comment (tiero) [reply internal/core/application/service.go:487]: @altafan

[the remark this answers is not in the document]
Comment (altafan) [reply internal/core/application/service.go:487]: @coderabbitai done
